### PR TITLE
watch services from all consul servers

### PR DESCRIPTION
**Pull request by uxgnaw**
When we have more than 200 services, we bump into consul leader cpu problems. Because services tags are too heavy, so I add stale option into watcher.go, this can get data from all servers.servers syncing data in 50ms, it's ok for us, because our service shutdown wait 5 sec.

### CODE REVIEW

**Comment by vtolstov** [comment]
may be best to add this stuff via WatchOption ?

**Comment by uxgnaw** [comment]
> may be best to add this stuff via WatchOption ?

AllowStale() function is already in option.go.
NewRegistry() function in consul.go is set allowStale is true for default.
I think it always should be true.

Also this problems is in "micro api" proxy, make it be a option, may will be modify micro?

**Comment by vtolstov** [comment]
i think that we need default like consul does. So NewRegistry() must check options and check AllowStale() if it true - use such option in watch too

**Comment by uxgnaw** [comment]
> i think that we need default like consul does. So NewRegistry() must check options and check AllowStale() if it true - use such option in watch too

ok, i will achieve it tomorrow

**Comment by uxgnaw** [comment]
i hava modify the code,  watch stale default is true same as consul new registry

File: registry/consul/options.go [hunk 1]
```diff
@@ -79,3 +79,18 @@ func TCPCheck(t time.Duration) registry.Option {
...
 	}
 }
+
+//
+//	This mode allows any server to service the read regardless of whether it is the leader.
+//	This function is as AllowStale but only for watch
+//
+// [1] https://www.consul.io/api/features/consistency#stale
+//
+func AllowWatchStale(v bool) registry.WatchOption {
```
Comment (vtolstov) [inline registry/consul/options.go:89]: we don't use string as key for context, use unexported struct like type watchStale struct{}

File: registry/consul/uitl.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package consul
```
Comment (vtolstov) [inline registry/consul/uitl.go:1]: file with name uitl must be util.go, but it not needed, check how options used from context in other places, we don't need additional file with additional function for single usage in one place

File: registry/consul/options.go (hunk 1 above)
Comment (uxgnaw) [reply registry/consul/options.go:91]: i fixed it

Comment (uxgnaw) [reply registry/consul/options.go:91]: it have fixed

File: registry/consul/uitl.go (hunk 2 above)
Comment (uxgnaw) [reply registry/consul/uitl.go]: i just find there two place to use it. what do you think how modify it?

**Comment by uxgnaw** [comment]
is there anything i can do?

**Pull request merged**